### PR TITLE
Implement logging telemetry log 1926 

### PR DESCRIPTION
**Pull request by pmoogi-redhat**
### Description
Story
As a member of the OpenShift Logging Team, I want telemetry metrics of the CLO
so that we have a better understanding of how CLF is being utilized by customers (e.g. which outputs)

Acceptance Criteria
Verifying telemetry metrics are available from the CLO metrics endpoint in prometheus

/cc @jcantrill @alanconway 
/assign 

### Links

- JIRA:  https://issues.redhat.com/browse/LOG-1926

### CODE REVIEW

**Review by alanconway — CHANGES_REQUESTED** [review]

File: internal/telemetry/clotelemetrymetric.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+func New() *CloTelemData {
+	return &CloTelemData{
+		CloInfo:       make(map[string]string),
+		ClfInputType:  make(map[string]string),
+		ClfOutputType: make(map[string]string),
+	}
+}
+
+var (
+	ClTelData = New()
```
[the remark this answers is not in the document]
Comment (pmoogi-redhat) [reply internal/telemetry/clotelemetrymetric.go]: @alanconway i see that EO also implemented by declaring global variables part of their metrics package. i wanted to do minimal changes to existing CLO stack and keep all new changes / implementation confining to this telemetry package. do you still think we should make this data a member of reconciler ? let me know.

[the remark this answers is not in the document]
Comment (alanconway) [reply internal/telemetry/clotelemetrymetric.go]: IMO this is a concurrent-unsafe pattern. Even if it is implemented safely today in EO and CLO,  it's much too easy for someone who doesn't understand the restrictions to make an unsafe reference to the map in new code..
Lets replace the map with a sync.Map for now, and please add a low-priority JIRA to review the pattern in future. Any variable with potentially concurre-unsafe use (i.e. not a constant or concurrent-safe type) should really be embedded in a type that will enforce (or at least document) the concurrency rules.

Here's a patch for a concurrent-safe string map:

```
diff --git a/internal/utils/stringmap.go b/internal/utils/stringmap.go
new file mode 100644
index 000000000..de1e10bc9
--- /dev/null
+++ b/internal/utils/stringmap.go
@@ -0,0 +1,28 @@
+package utils
+
+import "sync"
+
+// StringMap is a concurrent-safe map[string]string based on sync.Map
+type StringMap struct{ m sync.Map }
+
+// GetOK returns (value, true) if key is present and ("", false) if absent.
+func (m *StringMap) GetOK(key string) (s string, ok bool) {
+	x, ok := m.m.Load(key)
+	if ok {
+		s = x.(string)
+	}
+	return s, ok
+}
+
+// Get returns the value, or "" if the key is absent.
+func (m *StringMap) Get(key string) string {
+	if s, ok := m.GetOK(key); ok == true {
+		return s
+	}
+	return ""
+}
+
+// Set sets key to value.
+func (m *StringMap) Set(key, value string) {
+	m.m.Store(key, value)
+}
diff --git a/internal/utils/stringmap_test.go b/internal/utils/stringmap_test.go
new file mode 100644
index 000000000..ed008f6c7
--- /dev/null
+++ b/internal/utils/stringmap_test.go
@@ -0,0 +1,20 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStringMap(t *testing.T) {
+	m := StringMap{}
+	assert.Equal(t, "", m.Get("x"))
+	s, ok := m.GetOK("x")
+	assert.Equal(t, "", s)
+	assert.False(t, ok)
+	m.Set("x", "y")
+	assert.Equal(t, "y", m.Get("x"))
+	s, ok = m.GetOK("x")
+	assert.Equal(t, "y", s)
+	assert.True(t, ok)
+}
```

[the remark this answers is not in the document]
Comment (pmoogi-redhat) [reply internal/telemetry/clotelemetrymetric.go]: @alanconway resolved

**Review by alanconway — APPROVED** [review]
/lgtm

**Pull request merged**